### ISSUE TITLE
"copy-html" stage attempts to copy remote videos to local disk

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#2722]
- I believe this to be a bug, not a question about using DITA-OT.
- I read the [CONTRIBUTING][] file.

Let's say in my DITA topic I have a reference to a YouTube video like this:

        <object data="https://www.youtube.com/embed/GRxu4kwf0JI" outputclass="iframe"></object>

The "copy-image" stage actually attempts to download and copy these references locally:

     copy-html:
     [copy] Copying 9 resources to D:\projects\eXml\test\dita\samples\flowers\out\pdf-css
     [copy] Failed to copy https://www.youtube.com/embed/De2OhKvmqNI to D:\projects\eXml\test\dita\samples\flowers\out\pdf-css\https:\www.youtube.com\embed\De2OhKvmqNI due to java.io.FileNotFoundException D:\projects\eXml\test\dita\samples\flowers\out\pdf-css\https:\www.youtube.com\embed\De2OhKvmqNI (The filename, directory name, or volume label syntax is incorrect)
     [copy] Failed to copy https://www.youtube.com/embed/T-I1zjZ5DTc to D:\projects\eXml\test\dita\samples\flowers\out\pdf-css\https:\www.youtube.com\embed\T-I1zjZ5DTc due to java.io.FileNotFoundException D:\projects\eXml\test\dita\samples\flowers\out\pdf-css\https:\www.youtube.com\embed\T-I1zjZ5DTc (The filename, directory name, or volume label syntax is incorrect)

## Expected Behavior
The references to the remote placed videos should remain absolute and there should not be an attempt to download them locally.

## Actual Behavior
There is a failed attempt to download these references locally.

## Environment
Windows 10

* DITA-OT version:
  2.4.4
* Operating system and version _(Linux, macOS, Windows)_:
  Windows 10
* How did you run DITA-OT?
  oXygen, other editor, CMS, etc.
* Transformation type _(HTML5, PDF, custom, etc.)_: 
  XHTML

[CONTRIBUTING]: https://github.com/dita-ot/dita-ot/blob/develop/.github/CONTRIBUTING.md

**Comment by raducoravu** [comment]
Encountered this problem again as we have about 80 videos linked in our topics in the way I explained in the first post and on Linux for each video the target "copy-html" tries to download it locally because the references to videos have been marked in the .job.xml with format="html" .
Somehow the Java HTTP connection has problems with downloading YouTube resources and for each linked video we have a couple of minutes of timeout leading to the entire publishing process taking about 12 hours.

**Comment by raducoravu** [comment]
On the "org.dita.dost.reader.GenListModuleReader.parseObject(Attributes)" method somehow if the href is to an external location we should probably avoid adding it to the "nonConrefCopytoTargets" list.

**Comment by robander** [comment]
Need to double check if this is resolved in 3.0, but I'm guessing not.

**Comment by xephon2** [comment]
@raducoravu, is there a known workaround for 2.5.4?

**Comment by raducoravu** [comment]
Inside an Oxygen v20 installation there is a plugin which contributes our Java patches to the DITA OT:

Oxygen XML Editor 20\frameworks\dita\DITA-OT2.x\plugins\com.oxygenxml.dost.patches

so you could try to use that plugin which your DITA OT 2.5.4 as I created a patch on our side for the problem.

**Comment by xephon2** [comment]
@raducoravu, thank you, integrating your lib helped. It would be nice to have this patches in the core toolkit.

**Comment by robander** [comment]
After this came up in today's DITA-OT contributor call, I double checked and found that I still get the following failure with out-of-the-box 3.2.1:
```
 [gen-list] Type error at char 4 in xsl:apply-templates/@select on line 17 colum
n 43 of html2dita.xsl:
 [gen-list]   XTTE0570: A sequence of more than one item is not allowed as the v
alue of variable $dita
 [gen-list]   (<div/>, <script/>)
 [gen-list] [DOTJ013E][ERROR] Failed to parse the referenced file 'https://www.y
outube.com/embed/GRxu4kwf0JI'.: A sequence of more than one item is not allowed
as the value of variable $dita (<div/>, <script/>)
```

I also verified that as a workaround, you can switch from using `@data` to using `@datakeyref`, and defining the youtube link as a key (where you can legally specify the scope / format). I got a clean build (and working video link) by replacing this original markup:
`<object data="https://www.youtube.com/embed/GRxu4kwf0JI" outputclass="iframe"></object>`

with this object (in the topic):
`<object datakeyref="youtube" outputclass="iframe"/>`
and adding this key definition (in the map):
`<keydef keys="youtube" href="https://www.youtube.com/embed/GRxu4kwf0JI" scope="external" format="html"/>`

**Comment by raducoravu** [comment]
@robander As you suspected probably the "object" is treated as having scope="local". In my opinion, for "object" there should be some kind of auto-detect (as it does not have the @scope attribute by default). The auto-detect would first try to make a java.net.URL from the @data attribute value, if it does not succeed it should treat the scope as local, if it succeeds to create an URL from the @data value it should try to see if the URL protocol and host name is the same as the protocol and host name for the original DITA file which is being parsed. If it is the same, then again scope="local", otherwise scope="external".

**Comment by robander** [comment]
Fixed with #3306
